addons/nuxeo-retention/nuxeo-retention-core/src/main/java/org/nuxeo/retention/adapters/RetentionAdapterFactory.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * (C) Copyright 2019 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Guillaume RENARD
+ */
+package org.nuxeo.retention.adapters;
+
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.adapter.DocumentAdapterFactory;
+import org.nuxeo.retention.RetentionConstants;
+
+/**
+ * @since 11.1
+ */
+public class RetentionAdapterFactory implements DocumentAdapterFactory {
+
+    @Override
+    public Object getAdapter(DocumentModel doc, Class<?> itf) {
+        if (doc.hasFacet(RetentionConstants.RETENTION_RULE_FACET)) {
+            return new RetentionRule(doc);
+        } else if (doc.hasFacet(RetentionConstants.RECORD_FACET)) {
+            return new Record(doc);
+        } else {
+            return null;
+        }
```

Comment:
can be replaced by:
```
        if (doc.hasFacet(RetentionConstants.RETENTION_RULE_FACET)) {
            return new RetentionRule(doc);
        } else if (doc.hasFacet(RetentionConstants.RECORD_FACET)) {
            return new Record(doc);
        }
        return null;
```
to avoid the third branch

Reply:
This is 100% equivalent and generates the same bytecode, I'm not sure in what way it's better. It's just style.